src/controller/index.ts

```diff
@@ -131,32 +137,48 @@ export default class SimulariumController {
         return "";
     }
 
+    private createSimulatorConnection(
+        netConnectionConfig?: NetConnectionParams,
+        clientSimulatorParams?: ClientSimulatorParams,
+        localFile?: SimulariumFileFormat
+    ): void {
+        if (clientSimulatorParams) {
+            this.netConnection = new SimulatorConnection(clientSimulatorParams);
+        } else if (localFile) {
+            this.netConnection = new LocalFileConnection(
+                this.playBackFile,
+                localFile
+            );
+        } else if (netConnectionConfig) {
+            this.netConnection = new NetConnection(netConnectionConfig);
+        } else {
+            throw new Error(
+                "Insufficient data to determine and configure simulator connection"
+            );
+        }
+
+        this.netConnection.setTrajectoryFileInfoHandler(
+            (trajFileInfo: TrajectoryFileInfo) => {
+                this.handleTrajectoryInfo(trajFileInfo);
```

Comment:
Able to update TrajectoryFileInfo to latest version here (and just do it once here)??